wgengine/router/router_darwin.go

```diff
@@ -10,55 +10,10 @@ import (
 	"tailscale.com/types/logger"
 )
 
-type darwinRouter struct {
-	logf    logger.Logf
-	tunname string
-	Router
-}
-
 func newUserspaceRouter(logf logger.Logf, _ *device.Device, tundev tun.Device) (Router, error) {
-	tunname, err := tundev.Name()
-	if err != nil {
-		return nil, err
-	}
-
-	userspaceRouter, err := newUserspaceBSDRouter(logf, nil, tundev)
-	if err != nil {
-		return nil, err
-	}
-
-	return &darwinRouter{
-		logf:    logf,
-		tunname: tunname,
-		Router:  userspaceRouter,
-	}, nil
+	return newUserspaceBSDRouter(logf, nil, tundev)
```

Comment:
this `nil` is mysterious.

maybe `nil /* device */` or `(*device.Device)(nil)` ?

Or:

```go
    var nilDev *device.Device // nil, because Reasons Reasons
```

Reply:
Actually this was a strange extraneous assumption in existing code: newUserspaceBSDRouter doesn't do anything with that argument anyway. Fixed.